core/include/thread.h

```diff
@@ -507,13 +510,18 @@ void thread_print_stack(void);
  *
  * @param[in] thread    The thread to check for
  *
- * @return  `== 0`, if @p thread has no initialized message queue
- * @return  `!= 0`, if @p thread has its message queue initialized
+ * @pre     @p thread is not `NULL` and the thread corresponding to the thread
+ *          control block @p thread points to (still) exists
+ *
+ * @retval  false    @p thread has no initialized message queue
+ * @retval  true     @p thread has its message queue initialized
  */
-static inline int thread_has_msg_queue(const volatile struct _thread *thread)
+static inline bool thread_has_msg_queue(const thread_t *thread)
 {
+    assume(thread != NULL);
 #if defined(MODULE_CORE_MSG) || defined(DOXYGEN)
-    return (thread->msg_array != NULL);
+    const volatile thread_t *t = thread;
+    return (t->msg_array != NULL);
```

Comment:
Can you add a comment why the magic `volatile` is needed?

Reply:
I actually think it is not needed...

I'm just to scared that it might hide a bug. But maybe I should just be a bit more brave and drop it. I think this API only has three apps that use it, so I can just test them without the `volatile`.